server/src/main/java/org/opensearch/index/mapper/DocumentParser.java

```diff
@@ -206,6 +206,9 @@ private static MapperParsingException wrapInMapperParsingException(SourceToParse
     private static String[] splitAndValidatePath(String fullFieldPath) {
         if (fullFieldPath.contains(".")) {
             String[] parts = fullFieldPath.split("\\.");
+            if (parts.length == 0) {
+                throw new IllegalArgumentException("field name cannot contain only dot");
```

Comment:
This exception is propagated back to the user right? Should we say "." instead of "dot" then?

Reply:
Yes, I've changed that, please help to take a look, thank you!